components/codec/src/bits/mod.rs

```diff
@@ -0,0 +1,90 @@
+use std::{error, fmt};
+
+/// Bit
+///
+/// An enum used to represent a single bit, can be either `Zero` or `One`.
+#[derive(Debug, PartialEq)]
+pub enum Bit {
+    Zero,
+    One,
+}
+
+impl Bit {
+    /// Convert a bit to u64, so `Zero` becomes 0 and `One` becomes 1.
+    pub fn to_u64(&self) -> u64 {
+        match self {
+            Bit::Zero => 0,
+            Bit::One => 1,
+        }
+    }
+}
+
+/// Error
+///
+/// Enum used to represent potential errors when interacting with a stream.
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    Eof,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Eof => write!(f, "Encountered the end of the stream"),
+        }
+    }
+}
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        match *self {
+            Error::Eof => "Encountered the end of the stream",
+        }
+    }
+}
+
+/// Read
+///
+/// Read is a trait that encapsulates the functionality required to read from a
+/// stream of bytes.
+pub trait Read {
```

Comment:
The `Read` and `Write` seems unnecessary, and it also leads to confusion with the `std::io::Read` and `std::io::Write`.

Reply:
Remove useless trait `Read` and `Write`.